### PR TITLE
Add a UI test that verifies reload functionality

### PR DESCRIPTION
**Pull request by JohnSundell**
Add a UI test that verifies that a view can be rapidly reloaded without any problems.

To enable this test, the "Todo list" feature in the demo app now has a filter, and the delay of the search bar component can now be customized in a content operation.

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

File: demo/tests/ReloadUITests.swift [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        searchBar.typeText("s")
+        searchBar.typeText("t")
+        XCTAssertTrue(firstItem.exists)
+        XCTAssertFalse(secondItem.exists)
+        XCTAssertFalse(thirdItem.exists)
+    }
+    
+    private func addItem(named itemName: String) {
+        let app = XCUIApplication()
+        app.navigationBars["Todo List"].buttons["Add"].tap()
```
Comment (cerihughes) [inline demo/tests/ReloadUITests.swift]: Should probably assert that the button exists before we tap it.

File: demo/tests/ReloadUITests.swift [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        
+        let alert = app.alerts["Add an item"]
+        let textField = alert.textFields.element(boundBy: 0)
+        textField.tap()
```
Comment (cerihughes) [inline demo/tests/ReloadUITests.swift]: Assert this exists first

**Comment by JohnSundell** [comment]
@cerihughes Good points, makes sense to do asserts earlier, so that it's easier to debug the tests if something goes wrong 👍 Will fix!

**Comment by cerihughes** [comment]
🌳

**Pull request merged**